### ISSUE TITLE
ggpairs between two matrices or subsets

### ISSUE THREAD

**Issue by caterinap** [ggobi/ggally#137]
Thanks a lot for this very cool and useful package!!
I would have a suggestion for a potential future option for `ggpairs`.
Sometimes, and for large datasets, it might be useful to plot only some combinations of variables. For example all comparisons of a 1rst matrix VS a 2nd matrix (or 2 subsets of columns of a dataframe).
Something similar to what `TeachingDemos::pairs2` does but in a ggpairs style.

**Comment by schloerke** [comment]
Glad you like the package!

Ah! Makes sense.  

``ggpairs2``?  ``ggcombo``?

Should be like this one:
```{r}
TeachingDemos::pairs2( state.x77[,1:4], state.x77[,5:8], panel=panel.smooth)
```
![screenshot 2016-01-20 10 33 41](https://cloud.githubusercontent.com/assets/93231/12453464/51988f92-bf61-11e5-93b9-ce54b442fe04.png)

**Comment by schloerke** [comment]
Notes for implementation...

ggpairs2 would call ggpairs and then immediately call ``pm[cols1, cols2]``.

**Comment by caterinap** [comment]
That would be great! And thanks again for all your work!

**Comment by schloerke** [comment]
@caterinap , how do you imagine aesthetics working? (not a problem for the teaching demo)

Lets say I have a dataset X and Y.  If I want to color by column ColW, where should ggpairs2 listen?

I'm thinking dataset X.

Alternatively, It'd work great, if dataset Y was appended to dataset X, then there's one "data" that is used for all subplots, just each subplot changes their aes values. If column names clashed, they could use the columnLabels parameter.

Thoughts?

**Comment by schloerke** [comment]
current issues: 
* X, Y name clashing
  * if they clash, what should be presented? ClashName.1, OtherName.1
* where to grab aesthetics

....

Ah!
ggpairs2(data, columnsX, columnsY, ..., etc)

If the dataset given is user supplied.  this solves all issues as columns not in either columnsX or columnsY could be used for aesthetics.  

Will use this parameter idea.

**Comment by caterinap** [comment]
Oups sorry!! Just saw the message.
Yep the ggpairs2(data, columnsX, columnsY, col=ColourColumn, etc), is great and should also avoid problems of vectors of different length or NAs.

**Comment by schloerke** [comment]
Just posting an update... 

This is coming, but it'll take some time.  While subsetting from ggpairs makes sense at first, it doesn't make sense in what is displayed.  I'm rewriting the guts of ggpairs so that I can borrow code for ggduo.

It'll be called ggduo. :-)

**Comment by schloerke** [comment]
Project proposal for ggduo for Google Summer of Code: https://github.com/rstats-gsoc/gsoc2016/wiki/ggduo:-pairs-plots-for-multiple-regression,-cca,-time-series

**Comment by schloerke** [comment]
Released in v1.2.0
